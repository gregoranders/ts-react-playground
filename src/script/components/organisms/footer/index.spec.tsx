import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import * as TestSubject from './index';

describe(`${TestSubject.Footer.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(<TestComponent />);
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(<TestComponent />);
    const year = new Date().getFullYear();
    expect(testSubject.text()).toBe(`Â© ${year} by Gregor Anders`);
  });
});
