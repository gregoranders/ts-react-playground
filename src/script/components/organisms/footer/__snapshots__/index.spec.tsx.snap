// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot 1`] = `
<footer
  className="w3-padding-32 w3-center"
>
  Â© 
  2020
   by Gregor Anders
</footer>
`;
